File: ScaleformUI_Csharp/TimerBars/TimerBarPool.cs
namespace ScaleformUI
{
    public class TimerBarPool
    {
        private static List<TimerBarBase> _bars = new List<TimerBarBase>();

        public TimerBarPool()
        {
            _bars = new List<TimerBarBase>();
        }

        public List<TimerBarBase> ToList()
        {
            return _bars;
        }

        public void Add(TimerBarBase timer)
        {
            _bars.Add(timer);
        }

        public void Remove(TimerBarBase timer)
        {
            _bars.Remove(timer);
        }

        public void Draw()
        {
            int off = (Main.InstructionalButtons.ControlButtons != null && Main.InstructionalButtons.ControlButtons.Count != 0) || Main.InstructionalButtons.IsSaving ? 9 : 0;
            if (_bars.Count > 0)
            {
                for (int i = 0; i < _bars.Count; i++)
                {
                    _bars[i].Draw((i * 10) + off);
                }
            }
        }
    }
}

File: ScaleformUI_Lua/src/TimerBars/TimerBarPool.lua
TimerBarPool = setmetatable({}, TimerBarPool)
TimerBarPool.__index = TimerBarPool

---@class TimerBarPool
---@field public Bars table
---@field public New fun(self:TimerBarPool)
---@field public AddBar function
---@field public Draw function

function TimerBarPool.New()
    local _data = {
        Bars = {}
    }
    return setmetatable(_data, TimerBarPool)
end

function TimerBarPool:AddBar(timerBar)
    timerBar.Handle = #self.Bars + 1;
    self.Bars[timerBar.Handle] = timerBar
end

function TimerBarPool:RemoveBar(timerBar)    
    self.Bars[timerBar.Handle] = nil
end

function TimerBarPool:Draw()
    local offset = 0
    if #ScaleformUI.Scaleforms.InstructionalButtons.ControlButtons > 0 or ScaleformUI.Scaleforms.InstructionalButtons.IsSaving then
        offset = 9
    end
    for k, v in pairs(self.Bars) do
        v:Draw((k * 10) + offset)
    end
end
